File: DEMO-SERVER.Core/DAL/Entities/Common/BaseEntity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace DEMO.Core.DAL.Entities.Common
{
    public class BaseEntity : ICloneable
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long ID { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public string UpdatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; } = DateTime.Now;
        public RecordStatusType RecordStatus { get; set; } = RecordStatusType.Active;
        public string Remarks { get; set; }

        public object Clone()
        {
            return MemberwiseClone();
        }
        public enum RecordStatusType
        {
            Active = 0,
            Deleted = 1
        }
    }
}

File: Demo-App.web/src/core/model/BaseEntity.ts
import { RecordStatusType } from "src/core/enum/enum";

export class BaseEntity{
    ID: number = 0;
    RecordStatus: RecordStatusType | string = RecordStatusType.Active;
    CreatedOn: Date = new Date();
    CreatedBy: string = '';
    UpdatedOn: Date = new Date();
    UpdatedBy: string = '';
    Remarks: string = '';
}